examples/3D_contour.py

```diff
@@ -0,0 +1,165 @@
+"""
+A comprehensive example that shows the whole workflow of
+1) Loading data
+2) Defining the model structure for a joint distribution
+3) Estimating the parameter values of the model ("Fitting")
+4) Computing a 3D environmental contour
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from skimage.measure import marching_cubes
+import pandas as pd
+
+from virocon import (
+    read_ec_benchmark_dataset,
+    GlobalHierarchicalModel,
+    LogNormalDistribution,
+    ExponentiatedWeibullDistribution,
+    DependenceFunction,
+    WidthOfIntervalSlicer,
+    HighestDensityContour,
+    plot_marginal_quantiles,
+    plot_dependence_functions,
+)
+
+
+# Load sea state measurements. 
+
+data = pd.read_csv("datasets/NREL_data_oneyear.csv", sep=";", skipinitialspace=True)
+data.index = pd.to_datetime(data.pop(data.columns[0]), format="%Y-%m-%d-%H")
+
+
+# Define the structure of the joint model that we will use to describe
+# the the environmental data. To define a joint model, we define the
+# univariate parametric distributions and the dependence structure.
+# The dependence structure is defined using parametric functions.
+
+# A 3-parameter power function, which will be used as a dependence function.
+def _power3(x, a, b, c):
+    return a + b * x ** c
+
+# A 3-parameter exponential function, which will be used as a dependence function.
+def _exp3(x, a, b, c):
+    return a + b * np.exp(c * x)
+
+def _alpha3(x, a, b, c, d_of_x):
+    return (a + b * x ** c) / 2.0445 ** (1 / d_of_x(x))
+
+# A 4- parameter logistic function, which will be used as a dependence function.
+def _logistics4(x, a=1, b=1, c=-1, d=1):
+    return a + b / (1 + np.exp(c * (x - d)))
+
+
+# Lower and upper interval boundaries for the three parameter values.
+bounds = [(0, None), (0, None), (None, None)]
+logistics_bounds = [(0, None), (0, None), (None, 0), (0, None)]
+
+power3 = DependenceFunction(_power3, bounds, latex="$a + b * x^c$")
+exp3 = DependenceFunction(_exp3, bounds, latex="$a + b * \exp(c * x)$")
+logistics4 = DependenceFunction(_logistics4, logistics_bounds, 
+                                weights=lambda x, y: y, 
+                                latex="$a + b / (1 + \exp[c * (x -d)])$")
+alpha3 = DependenceFunction(_alpha3, bounds, d_of_x=logistics4, 
+                               weights=lambda x, y: y,
+                               latex="$(a + b * x^c) / 2.0445^{1 / F()}$")
+
+# Set up distribution functions for the envrionmental variables.
```

Comment:
I think a better formulation is "Define the structure of the joint distribution."